src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -502,6 +511,7 @@ def __deepcopy__(self, memodict={}) -> "SnowflakePlan":  # noqa: B006
         copied_plan._is_valid_for_replacement = True
         if copied_source_plan:
             copied_source_plan._is_valid_for_replacement = True
+        copied_plan.uuid = self.uuid
```

Comment:
by definition, the copied_plan should have a different uuid here, right? because it is a different node